### PR TITLE
fix(demo): make the music player correctly work with v9

### PR DESCRIPTION
**Pull request by kisvegabor**
### Description of the feature or fix

cc @espzav 

### Notes
- Update the [Documentation](https://github.com/lvgl/lvgl/tree/master/docs) if needed.
- Add [Examples](https://github.com/lvgl/lvgl/tree/master/examples) if relevant.
- Add [Tests](https://github.com/lvgl/lvgl/blob/master/tests/README.md) if applicable.
- If you added new options to `lv_conf_template.h` run [lv_conf_internal_gen.py](https://github.com/lvgl/lvgl/blob/master/scripts/lv_conf_internal_gen.py) and update [Kconfig](https://github.com/lvgl/lvgl/blob/master/Kconfig).
- Run `scripts/code-format.py` ([astyle](http://astyle.sourceforge.net/install.html) version [v3.4.12](https://github.com/szepeviktor/astyle/releases/tag/v3.4.12) needs to be installed) and follow the [Code Conventions](https://docs.lvgl.io/master/CODING_STYLE.html).
- Mark the Pull request as [Draft](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/changing-the-stage-of-a-pull-request) while you are working on the first version, and mark is as _Ready_ when it's ready for review.
- When changes were requested, [re-request review](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/requesting-a-pull-request-review) to notify the maintainers.
- Help us to review this Pull Request! Anyone can [approve or request changes](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/reviewing-changes-in-pull-requests/approving-a-pull-request-with-required-reviews).

### CODE REVIEW

**Comment by lhdjply** [comment]
The playback effect is still a bit strange

image

**Comment by kisvegabor** [comment]
Thanks, should be fixed!

**Comment by espzav** [comment]
@kisvegabor please, on the small screen (M5Stack, ESP-BOX, ...), there is not visible whole LVGL logo on the start. The first "L" is missing.

**Comment by espzav** [comment]
> @kisvegabor please, on the small screen (M5Stack, ESP-BOX, ...), there is not visible whole LVGL logo on the start. The first "L" is missing.

I am sorry, I tested bad repository. This looks good, but it crashes during some animation. I will continue test it tomorrow.

File: demos/music/lv_demo_music_main.c [hunk 1]
```diff
@@ -842,7 +851,7 @@ static void spectrum_draw_event_cb(lv_event_t * e)
         }
 
         const int32_t amax = 20;
-        int32_t animv = spectrum_i - spectrum_lane_ofs_start;
+        int32_t animv = spectrum_i - 0;
```
Comment (lhdjply) [inline demos/music/lv_demo_music_main.c:849]: There is no need to reduce 0 here

**Comment by espzav** [comment]
When enabled `LV_USE_PERF_MONITOR` it crashes. Here is a log:
```
Guru Meditation Error: Core  1 panic'ed (LoadProhibited). Exception was unhandled.

Core  1 register dump:
PC      : 0x4202460e  PS      : 0x00060330  A0      : 0x82014a91  A1      : 0x3fcb3a10
0x4202460e: auto_step_cb at C:/WORK/GIT/esp-bsp-01/examples/display_lvgl_demos/components/lvgl__lvgl/demos/music/lv_demo_music.c:234

A2      : 0x3fcd3424  A3      : 0x4202448c  A4      : 0x00060d20  A5      : 0x3fca7694
0x4202448c: auto_step_cb at C:/WORK/GIT/esp-bsp-01/examples/display_lvgl_demos/components/lvgl__lvgl/demos/music/lv_demo_music.c:161

A6      : 0x3fccce54  A7      : 0x3fca4624  A8      : 0x8202460e  A9      : 0x3fcb39f0
A10     : 0x00000000  A11     : 0x3c0788b8  A12     : 0x00000000  A13     : 0x0000005a
A14     : 0x00000000  A15     : 0x00000013  SAR     : 0x00000020  EXCCAUSE: 0x0000001c
EXCVADDR: 0x00000050  LBEG    : 0x400570e8  LEND    : 0x400570f3  LCOUNT  : 0x00000000
0x400570e8: memset in ROM
0x400570f3: memset in ROM



Backtrace: 0x4202460b:0x3fcb3a10 0x42014a8e:0x3fcb3a40 0x42014be9:0x3fcb3a70 0x4200a5c9:0x3fcb3a90
0x4202460b: auto_step_cb at C:/WORK/GIT/esp-bsp-01/examples/display_lvgl_demos/components/lvgl__lvgl/demos/music/lv_demo_music.c:233
0x42014a8e: lv_timer_exec at C:/WORK/GIT/esp-bsp-01/examples/display_lvgl_demos/components/lvgl__lvgl/src/misc/lv_timer.c:300 (discriminator 2)
 (inlined by) lv_timer_handler at C:/WORK/GIT/esp-bsp-01/examples/display_lvgl_demos/components/lvgl__lvgl/src/misc/lv_timer.c:105 (discriminator 2)
0x42014be9: lv_timer_handler at C:/WORK/GIT/esp-bsp-01/examples/display_lvgl_demos/components/lvgl__lvgl/src/misc/lv_timer.c:71
0x4200a5c9: lvgl_port_task at C:/WORK/GIT/esp-bsp-01/components/esp_lvgl_port/src/lvgl9/esp_lvgl_port.c:242
```

It was working on last released LVGL. Only this branch crashes.

I have another issue with this demo (on release too), task watchdog is triggered. It can be my problem, this problem is only on small screens (ESP-BOX, M5Stack, ..). For ESP32-s3-EV board with RGB screen it is ok.

**Comment by kisvegabor** [comment]
> When enabled LV_USE_PERF_MONITOR it crashes. 

I've fixed it by rebasing on `master`. 

> I have another issue with this demo (on release too), task watchdog is triggered. It can be my problem, this problem is only on small screens (ESP-BOX, M5Stack, ..). For ESP32-s3-EV board with RGB screen it is ok.

Let me know if I can help with that.

**Comment by espzav** [comment]
> I've fixed it by rebasing on `master`.

Thank you, it's working now. 

> Let me know if I can help with that.

Seems that I have some issue in LVGL port task. When I disable touch interrupt (it menas that LV_INDEV_MODE_EVENT is disabled for indev), it is ok. I am not using touch during the demo - read not called in the task.

**Comment by espzav** [comment]
Please, where is difference, when I put this line to initialization indev?
```
lv_indev_set_mode(indev, LV_INDEV_MODE_EVENT);
```

When I use it, it shows errors and warnings like these:
```
E (32376) task_wdt: Task watchdog got triggered. The following tasks/users did not reset the watchdog in time:
E (32376) task_wdt:  - IDLE0 (CPU 0)
E (32376) task_wdt: Tasks currently running:
E (32376) task_wdt: CPU 0: LVGL task

Backtrace: 0x4203D12A:0x3FCA1C90 0x40382811:0x3FCA1CC0 0x4202C77C:0x3FCB3940 0x4202CF6E:0x3FCB3960 0x4202D7FB:0x3FCB3990 0x42014C0E:0x3FCB3A20 0x42014D6D:0x3FCB3A50 0x4200A59D:0x3FCB3A70
0x4203d12a: task_wdt_timeout_handling at C:/WORK/GIT/esp-idf_GitLab/components/esp_system/task_wdt/task_wdt.c:434
 (inlined by) task_wdt_isr at C:/WORK/GIT/esp-idf_GitLab/components/esp_system/task_wdt/task_wdt.c:507
0x40382811: _xt_lowint1 at C:/WORK/GIT/esp-idf_GitLab/components/xtensa/xtensa_vectors.S:1240
0x4202c77c: wait_for_flushing at C:/WORK/GIT/esp-bsp-01/examples/display_lvgl_demos/components/lvgl__lvgl/src/core/lv_refr.c:1089 (discriminator 1)
0x4202cf6e: refr_area_part at C:/WORK/GIT/esp-bsp-01/examples/display_lvgl_demos/components/lvgl__lvgl/src/core/lv_refr.c:674
0x4202d7fb: refr_area at C:/WORK/GIT/esp-bsp-01/examples/display_lvgl_demos/components/lvgl__lvgl/src/core/lv_refr.c:661
 (inlined by) refr_invalid_areas at C:/WORK/GIT/esp-bsp-01/examples/display_lvgl_demos/components/lvgl__lvgl/src/core/lv_refr.c:562
 (inlined by) _lv_display_refr_timer at C:/WORK/GIT/esp-bsp-01/examples/display_lvgl_demos/components/lvgl__lvgl/src/core/lv_refr.c:374
0x42014c0e: lv_timer_exec at C:/WORK/GIT/esp-bsp-01/examples/display_lvgl_demos/components/lvgl__lvgl/src/misc/lv_timer.c:305 (discriminator 2)
 (inlined by) lv_timer_handler at C:/WORK/GIT/esp-bsp-01/examples/display_lvgl_demos/components/lvgl__lvgl/src/misc/lv_timer.c:107 (discriminator 2)
0x42014d6d: lv_timer_handler at C:/WORK/GIT/esp-bsp-01/examples/display_lvgl_demos/components/lvgl__lvgl/src/misc/lv_timer.c:71
0x4200a59d: lvgl_port_task at C:/WORK/GIT/esp-bsp-01/components/esp_lvgl_port/src/lvgl9/esp_lvgl_port.c:242
```

**Comment by espzav** [comment]
> It seems it stuck in `wait_for_flushing` so maybe `lv_display_flush_ready` is not called for some reason. In `LV_INDEV_MODE_EVENT` LVGL won't call the indev's `read_cb` periodically but you need call `lv_indev_read(my_indev);` when an indev event (press, release, move, etc) happened.

It is not stucked too long, only more time than watchdog is set. The demo is working. 
I know, that I should call `lv_indev_read` when `LV_INDEV_MODE_EVENT` is set. I call it only when interrupt from touch was received. In my case, this line is only reason, why I got watchdog errors. When I set `LV_INDEV_MODE_EVENT`, there are watchdog errors. I fixed it.

**Comment by kisvegabor** [comment]
@espzav 
Do you see any other issues, or is ready to merge?

**Comment by espzav** [comment]
> @espzav Do you see any other issues, or is ready to merge?
No, I didn't find any other issues. You can merge it.

**Review by XuNeo — DISMISSED** [review]

**Review by liamHowatt — COMMENTED** [review]
2 issues I found but couldn't root-cause.

Only when `LV_DEMO_MUSIC_ROUND` is `1`.
image

Only when `LV_DEMO_MUSIC_LARGE` is `1` (for any display size).
image

**Comment by kisvegabor** [comment]
Thanks for pointing these out Liam! 

> Only when LV_DEMO_MUSIC_ROUND is 1.

It's the intended behavior because on round screens the selected item should be in the middle to make sure it's not clipped by the shape of the display. 

> Only when LV_DEMO_MUSIC_LARGE is 1 (for any display size).

I've just fixed it.

**Review by liamHowatt — DISMISSED** [review]
> It's the intended behavior because on round screens the selected item should be in the middle to make sure it's not clipped by the shape of the display.

Ahh of course.

> I've just fixed it.

Nice, thanks!

File: src/misc/lv_anim.c [hunk 2]
```diff
@@ -274,45 +274,46 @@ int32_t lv_anim_path_overshoot(const lv_anim_t * a)
 int32_t lv_anim_path_bounce(const lv_anim_t * a)
 {
     /*Calculate the current step*/
-    int32_t t = lv_map(a->act_time, 0, a->duration, 0, LV_BEZIER_VAL_MAX);
+    int32_t t = lv_map(a->act_time, 0, a->duration, 0, 1024);
```
Comment (FASTSHIFT) [inline src/misc/lv_anim.c]: Why change this `LV_BEZIER_VAL_MAX` -> 1024?

Comment (kisvegabor) [reply src/misc/lv_anim.c]: Thanks, fixed!

**Review by liamHowatt — DISMISSED** [review]

File: demos/music/lv_demo_music_main.c [hunk 3]
```diff
@@ -290,15 +293,18 @@ lv_obj_t * _lv_demo_music_main_create(lv_obj_t * parent)
     lv_obj_t * logo = lv_image_create(lv_screen_active());
     lv_image_set_src(logo, &img_lv_demo_music_logo);
     lv_obj_move_foreground(logo);
+    lv_obj_align_to(logo, spectrum_obj, LV_ALIGN_CENTER, 0, 0);
 
+#if LV_DEMO_MUSIC_SQUARE != 0 && LV_DEMO_MUSIC_ROUND != 0
```
Comment (XuNeo) [inline demos/music/lv_demo_music_main.c]: Please check if you mean `#if LV_DEMO_MUSIC_SQUARE || LV_DEMO_MUSIC_ROUND`

Comment (kisvegabor) [reply demos/music/lv_demo_music_main.c]: It was really incorrect. The title should be created only for non square and non round display.

**Review by liamHowatt — APPROVED** [review]

**Review by XuNeo — APPROVED** [review]

**Pull request merged**